src/aggregations/bucket-aggregations/auto-histogram-aggregation-base.js

```diff
@@ -0,0 +1,201 @@
+} = require('../../core');
+
+const BucketAggregationBase = require('./bucket-aggregation-base');
+
+const invalidDirectionParam = invalidParam('', 'direction', "'asc' or 'desc'");
+
+/**
+ * The `AutoHistogramAggregationBase` provides support for common options used across
+ * various auto-histogram `Aggregation` implementations like
+ * Auto Date Histogram aggregation.
+ *
+ * **NOTE:** Instantiating this directly should not be required.
+ * However, if you wish to add a custom implementation for whatever reason,
+ * this class could be extended.
+ *
+ * @param {string} name The name which will be used to refer to this aggregation.
+ * @param {string} aggType Type of aggregation
+ * @param {string=} field The field to aggregate on
+ * @param {number=} bucket count to generate histogram over.
+ *
+ * @extends BucketAggregationBase
+ */
+class AutoHistogramAggregationBase extends BucketAggregationBase {
+    // eslint-disable-next-line require-jsdoc
+    constructor(name, aggType, field, buckets) {
+        super(name, aggType, field);
+
+        if (!isNil(buckets)) this._aggsDef.buckets = buckets;
+    }
+
+    /**
+     * Sets the histogram bucket count. Buckets are generated based on this value.
+     *
+     * @param {number=} bucket count to generate histogram over.
+     * @returns {HistogramAggregationBase} returns `this` so that calls can be chained
+     */
+    buckets(buckets) {
+        this._aggsDef.buckets = buckets;
+        return this;
+    }
+
+    /**
+     * Sets the format expression for `key_as_string` in response buckets.
+     * If no format is specified, then it will use the first format specified in the field mapping.
+     *
+     * @example
+     * const agg = esb.dateHistogramAggregation(
+     *     'sales_over_time',
+     *     'date',
+     *     '1M'
+     * ).format('yyyy-MM-dd');
+     *
+     * @param {string} fmt Format mask to apply on aggregation response. Example: ####.00.
+     * For Date Histograms, supports expressive [date format pattern](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-daterange-aggregation.html#date-format-pattern)
+     * @returns {HistogramAggregationBase} returns `this` so that calls can be chained
+     */
+    format(fmt) {
+        this._aggsDef.format = fmt;
+        return this;
+    }
+
+    /**
+     * The offset parameter is used to change the start value of each bucket
+     * by the specified positive (+) or negative offset (-).
+     * Negative offset is not applicable on HistogramAggregation.
+     * In case of DateHistogramAggregation, duration can be
+     * a value such as 1h for an hour, or 1d for a day.
+     *
+     * @example
+     * const agg = esb.dateHistogramAggregation('by_day', 'date', 'day').offset('6h');
+     *
+     * @param {string} offset Time or bucket key offset for bucketing.
+     * @returns {HistogramAggregationBase} returns `this` so that calls can be chained
+     */
+    offset(offset) {
```

Comment:
I don't think this is supported on `auto_date_histogram`.

Reply:
checked with docs and only add the proper params